### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by ilkhomibodillayev** [SocialbitGmbH/SwiftAddressBook#101]
I am getting crash on:
fileprivate func convertNSDictionary<T : NSCopying, U, V, W>(_ d : NSDictionary?, keyConverter : (T) -> V, valueConverter : (U) -> W ) -> Dictionary<V, W>? {
		if let d2 = d {
			var dict = Dictionary<V,W>()
			for key in d2.allKeys as! Array<T> {
				let newKey = keyConverter(key)
				**let newValue = valueConverter(d2[key] as! U)**
				dict[newKey] = newValue
			}
			return dict
		}
		else {
			return nil
		}
	}
Any suggestion for me please?

**Comment by TAKeanice** [comment]
Yes, please state which version of iOS, Xcode, MacOS, for which version of iOS you compile, and which version of SwiftAddressBook you are using. I guess the line you marked **like this** is the one that the crash is happening in. Probably the fault is the forced cast. That means the dictionary contains some value of _not the type U_ whichever that type is. If possible please inspect the crash (by setting an exception breakpoint) and tell me which type it is. Now tell me how it came in there. Good luck!
